Code/GraphMol/FileParsers/MolFileParser.cpp

```diff
@@ -765,6 +767,55 @@ void ParseZBOLine(RWMol *mol, const std::string &text, unsigned int line) {
   }
 }
 
+void ParseMarvinSmartsLine(RWMol *mol, const std::string &text, unsigned int line) {
+  const unsigned int atomNumStart = 10;
+  const unsigned int smartsStart = 15;
+  const unsigned int SMA = 7;
+  // M  MRV SMA   1 [*;A]
+  // 01234567890123456789
+  //           1111111111
+  if (text.substr(SMA, 3) != "SMA") {
+    return;
+  }
+
+  unsigned int idx;
+  std::string idxTxt = text.substr(atomNumStart, smartsStart-atomNumStart);
+  try {
+    idx = FileParserUtils::stripSpacesAndCast<unsigned int>(idxTxt) - 1;
+  } catch (boost::bad_lexical_cast &) {
+        std::ostringstream errout;
+        errout << "Cannot convert " << idxTxt << " to an atom index on line "
+               << line;
+        throw FileParseException(errout.str());
+  }
+
+  URANGE_CHECK(idx, mol->getNumAtoms());
+  // Should we check the validity of the marvin line here?  Should we automatically
+  //   Add these as recursive smarts?  I tend to think so...
+  std::string sma = text.substr(smartsStart);
+  Atom * at = mol->getAtomWithIdx(idx);
+  at->setProp(common_properties::MRV_SMA, sma);
+  RWMol *m = 0;
+  try {
+    m = SmartsToMol(sma);
+  } catch(...) {
+  }
```

Comment:
hmm, not sure how I feel about just ignoring the failure. Are you doing this because there's some reasonable scenario where the value of this property isn't a SMARTS, despite the name?

If not, I think it's worth throwing a `FileParseException` here.